providers/implementations/keymgmt/ml_kem_kmgmt.c

```diff
@@ -362,12 +403,13 @@ static int ml_kem_import(void *vkey, int selection, const OSSL_PARAM params[])
 
     include_private = selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY ? 1 : 0;
     res = ml_kem_key_fromdata(key, params, include_private);
-#ifdef FIPS_MODULE
     if (res > 0 && include_private && !ml_kem_pairwise_test(key)) {
+#ifdef FIPS_MODULE
         ossl_set_error_state(OSSL_SELF_TEST_TYPE_PCT);
+#endif
+        ossl_ml_kem_key_reset(key);
         res = 0;
     }
```

Comment:
```suggestion
#if defined(FIPS_MODULE) || defined(OPENSSL_POINTLESS_LAMP_PCT)
    if (res > 0 && include_private && !ml_kem_pairwise_test(key)) {
        ossl_set_error_state(OSSL_SELF_TEST_TYPE_PCT);
        ossl_ml_kem_key_reset(key);
        res = 0;
    }
```

Reply:
I'd much prefer to see no performance impact.  This is is technically useless.  Politically beneficial tests are best when omitted.